### PR TITLE
Fixes #1046, ensures that flextesa works without CORS issues

### LINKED ISSUE #1046: [Bug]: CORS issue trying to connect to sandbox via Temple
### What happened?

When trying to connect to sandbox via Temple getting an issue with CORS
`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:20000/chains/main/chain_id. (Reason: CORS header 'Access-Control-Allow-Origin' missing). Status code: 200.`

### What should have happened?

There should be no errors trying to connect to sandbox from any application because environment variable should be set up to *

### Steps to reproduce the bug

Steps to reproduce the error:

### Relevant log output

_No response_

### Screenshots and images

Screenshot and images

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mweichert**
# 🏗️ PR ➾ 1296 - Ensure that flextesa does not produce CORS issues

Fixes #1046 

## 🪂 Pre-Merge Checklist

- [x] 🐬 I have commented my code, particularly in hard-to-understand areas
- [x] 🦀 Automated tests have been written and added to this PR
- [x] 🤿 Corresponding changes have been made to all documentation
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [x] 🔱 The test plan has been implemented and verified by an SDET
- [x] 🏄‍♂️ Myself and others have built this branch and done manual testing on the change
- [x] ⛵ A summary describing the change has been written for release notes

----------------------------------------------------------------------------------------------------------------------------

## 🪁 Description

When running a dapp locally, and trying to connect it to a flextesa sandbox, users were getting network failures due to CORS issues.

This was fixed previously, but broke when the use of `execa` was introduced in the @taqueria/flextesa-manager package.

This is necessary for my effort to ensure for testing beacon-sdk with flextesa sandboxes works correctly: https://github.com/ecadlabs/taqueria-beacon-sandbox

## 🎢 Test Plan

I will be writing a test that uses fetch to make a cross-origin request and assure that it succeeds, thus - not having CORS issues.

----------------------------------------------------------------------------------------------------------------------------

### 🛸 Type of Change

- [ ] 🐟 Minor change (non-breaking change of very limited scope)
- [x] 🦑 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🌊 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🐳 Major refactor (breaking changes and significant impact to the codebase)

----------------------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: taqueria-plugin-taquito/originate.ts [hunk 1]
```diff
@@ -165,20 +165,64 @@ const createBatch = async (parsedArgs: Opts, tezos: TezosToolkit, destination: s
 		if (error.message) {
 			const msg = error.message;
 			if (/ENOTFOUND/.test(msg)) {
-				sendErr(msg + ' - The RPC URL may be invalid. Check your ./taq/config.json.');
+				sendErr('The RPC URL may be invalid. Check your ./taq/config.json.\n');
```
Comment (AlirezaHaghshenas) [inline taqueria-plugin-taquito/originate.ts:168]: Adding the JSON path might help the user

Comment (AlirezaHaghshenas) [inline taqueria-plugin-taquito/originate.ts:168]:
```suggestion
				sendErr('The RPC URL may be invalid. Check your ./.taq/config.json.\n');
```
Also, adding the JSON path might help the user

**Review by AlirezaHaghshenas — APPROVED** [review]

**Review by hu3man — APPROVED** [review]

File: tests/e2e/taqueria-plugin-flextesa.spec.mts [hunk 2]
```diff
@@ -176,6 +177,11 @@ describe('E2E Testing for taqueria flextesa plugin sandbox starts/stops', () =>
 		}
 	});
 
+	test.only('Verify that Taqueria accepts any origin and does not emit any CORS related errors', async () => {
```
Comment (alexzbusko) [inline tests/e2e/taqueria-plugin-flextesa.spec.mts:180]: This only should be removed

Comment (mweichert) [reply tests/e2e/taqueria-plugin-flextesa.spec.mts:180]: Ah, thanks - it snuck in there while debugging the node-fetch issue I mentioned.

**Comment by hu3man** [comment]
Seems to be working great!

@mweichert , can you please slice out the ticket for the testing re-work and then we can merge this?

**Comment by mweichert** [comment]
@hu3man I'm actually working on this right now. I think what I'll do is provide a test using a library that works around the test framework issues, just to unblock this. 

This will ensure that we can work on the test framework adjustments without feeling too much pressure as well.

**Comment by hu3man** [comment]
Awesome, sounds great @mweichert!

File: taqueria-plugin-taquito/transfer.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ const configureToolKitWithSandbox = (parsedArgs: Opts, sandboxName: string): Pro
 	tezos.setProvider({
 		signer: new InMemorySigner((defaultAccount.secretKey as string).replace(/^unencrypted:/, '')),
 	});
-	return tezos;
+	return Promise.resolve(tezos);
```
Comment (jchenche) [inline taqueria-plugin-taquito/transfer.ts:68]: Why do we need this? The function is declared as `async` so isn't this essentially the same as `return tezos`

File: taqueria-plugin-taquito/originate.ts [hunk 4]
```diff
@@ -184,9 +186,9 @@ const createBatch = async (parsedArgs: Opts, tezos: TezosToolkit, destination: s
 				}
 			} else {
 				sendErr(
-					msg
-						+ " - There was a problem communicating with the chain. Check the RPC URL of the network or sandbox you're targeting in config.json.\n",
+					"There was a problem communicating with the chain. Perhaps review your RPC URL of the network or sandbox you're targeting.\n",
```
Comment (hu3man) [inline taqueria-plugin-taquito/originate.ts:189]: Let's avoid the use of weak/indirect language like 'perhaps'

Also, the e2e tests need to be updated as they fail when trying to match this output string

**Review by hu3man — CHANGES_REQUESTED** [review]

File: taqueria-plugin-taquito/transfer.ts (hunk 3 above)
Comment (mweichert) [reply taqueria-plugin-taquito/transfer.ts:68]: Semantically, there's no difference here. However, `tsup` doesn't warn about an async without an await.

File: taqueria-plugin-taquito/originate.ts (hunk 4 above)
Comment (mweichert) [reply taqueria-plugin-taquito/originate.ts:189]: Hmm, odd - this looks like another merge issue. That makes me nervous. I didn't purposely revert the messaging (that I added in a previous PR)

Comment (mweichert) [reply taqueria-plugin-taquito/originate.ts]: I've restored the original msging and adjusted the last msg to be stronger.

File: taqueria-plugin-taquito/transfer.ts (hunk 3 above)
Comment (jchenche) [reply taqueria-plugin-taquito/transfer.ts:68]: What do you mean?

You mean to say "`tsup` does warn about an async without an await."?

File: tests/jest.config.e2e.ts [hunk 5]
```diff
@@ -185,7 +185,7 @@ export default {
 	// ],
 
 	// Changing default timeout from 5000 to 30000 for tests
-	testTimeout: 300000,
+	testTimeout: 180000,
```
Comment (alexzbusko) [inline tests/jest.config.e2e.ts:188]: We set this to be higher for the scaffolding tests running in the pipeline. They run a lot slower there which causes the test to timeout and report a failure

**Review by hu3man — APPROVED** [review]

**Pull request merged**